pkg/apis/autoscaling/v1/fleetautoscaler.go

```diff
@@ -195,6 +210,61 @@ type ListPolicy struct {
 	BufferSize intstr.IntOrString `json:"bufferSize"`
 }
 
+// Between defines the time period that the policy is eligible to be applied.
+type Between struct {
+	// Start is the datetime that the policy is eligible to be applied.
+	// This field must conform to RFC3339 format. If this field not set or is in the past, the policy is eligible to be applied
+	// as soon as the fleet autoscaler is running.
+	Start string `json:"start"`
+
+	// End is the datetime that the policy is no longer eligible to be applied.
+	// This field must conform to RFC3339 format. If not set, the policy is always eligible to be applied, after the start time above.
+	End string `json:"end"`
+}
+
+// ActivePeriod defines the time period that the policy is applied.
+type ActivePeriod struct {
+	// Timezone to be used for the startCron field. If not set, startCron is defaulted to the UTC timezone.
+	Timezone string `json:"timezone"`
+
+	// StartCron defines when the policy should be applied.
+	// If not set, the policy is always to be applied within the start and end time.
+	// This field must conform to UNIX cron syntax.
+	StartCron string `json:"startCron"`
+
+	// Duration is the length of time that the policy is applied.
+	// If not set, the duration is indefinite.
+	// A duration string is a possibly signed sequence of decimal numbers,
+	// (e.g. "300ms", "-1.5h" or "2h45m").
+	// The representation limits the largest representable duration to approximately 290 years.
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	Duration string `json:"duration"`
+}
+
+// Schedule defines when the policy should be applied.
+type Schedule struct {
+	// Between defines the time period that the policy is eligible to be applied.
+	Between Between `json:"between"`
+
+	// ActivePeriod defines the time period that the policy is applied.
+	ActivePeriod ActivePeriod `json:"activePeriod"`
+}
+
+// ChainEntry defines a single entry in the ChainPolicy.
+type ChainEntry struct {
+	// ID is the unique identifier for a ChainEntry. If not set the identifier will be set to the index of chain entry.
+	ID string `json:"id"`
+
+	// Schedule defines when the policy is applied.
+	Schedule Schedule `json:"schedule"`
```

Comment:
This feels wrong to me. I think this is going to be an embedded `FleetAutoscalerPolicy` instead, since we don't have a top level `scheduled` within `ChainEntry`, you have a `type` first to say if the chain entry is a scheduled one or not.

Reply:
I've went ahead and separated `Schedule` from `Chain`, and it's now it's own policy.